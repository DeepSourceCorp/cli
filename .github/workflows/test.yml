name: Unit Tests

on:
  push:
    branches: [develop]
  pull_request:
    branches: [develop]

jobs:
  test:
    runs-on: ubuntu-latest
    env:
      NODE_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}
      ZEAL_SECRET: ${{secrets.GITHUB_TOKEN}}

    steps:
      - name: Checkout Bifrost
        uses: actions/checkout@v2

      - name: Setting Up Node
        uses: actions/setup-node@v2
        with:
          node-version: '14.x'
          always-auth: true
          registry-url: https://npm.pkg.github.com/
          scope: '@deepsourcelabs'
          cache: 'yarn'

      - name: Install packages
        run: yarn

      - name: Run Test
        run: yarn test --coverage=true --coverageReporters=cobertura

      - name: Install DeepSource CLI
        run: curl https://deepsource.io/cli | sh

      - name: Set DEEPSOURCE_DSN
        run: |
          ./bin/deepsource report --analyzer test-coverage --key javascript --value-file ./coverage/cobertura-coverage.xml
        env:
          DEEPSOURCE_DSN: ${{ secrets.BIFROST_DSN }}
