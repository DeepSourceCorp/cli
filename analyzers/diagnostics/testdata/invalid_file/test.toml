name = "2do-Checker"
