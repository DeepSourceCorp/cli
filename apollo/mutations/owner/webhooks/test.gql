mutation testWebhook($webhookId: ID!) {
  testWebhook(input: { webhookId: $webhookId }) {
    ok
  }
}
