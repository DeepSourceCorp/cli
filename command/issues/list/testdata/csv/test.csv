analyzer,issue_code,issue_title,occurence_title,issue_category,path,begin_line,begin_column,end_line,end_column
go,RVV-B0013,Unused method receiver detected,Unused method receiver detected,,deepsource/transformers/queries/get_transformers.go,34,0,34,0
go,RVV-B0013,Unused method receiver detected,Unused method receiver detected,,deepsource/transformers/queries/get_transformers.go,44,0,44,0
