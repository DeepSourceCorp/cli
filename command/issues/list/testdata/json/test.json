{
	"occurences": [
		{
			"analyzer": "go",
			"issue_code": "RVV-B0013",
			"issue_title": "Unused method receiver detected",
			"occurence_title": "Unused method receiver detected",
			"issue_category": "",
			"location": {
				"path": "deepsource/transformers/queries/get_transformers.go",
				"position": {
					"begin": {
						"line": 34,
						"column": 0
					},
					"end": {
						"line": 34,
						"column": 0
					}
				}
			}
		},
		{
			"analyzer": "go",
			"issue_code": "RVV-B0013",
			"issue_title": "Unused method receiver detected",
			"occurence_title": "Unused method receiver detected",
			"issue_category": "",
			"location": {
				"path": "deepsource/transformers/queries/get_transformers.go",
				"position": {
					"begin": {
						"line": 44,
						"column": 0
					},
					"end": {
						"line": 44,
						"column": 0
					}
				}
			}
		}
	],
	"summary": {
		"total_occurences": 2,
		"unique_issues": 1
	}
}