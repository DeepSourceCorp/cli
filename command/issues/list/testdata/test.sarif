{
  "version": "2.1.0",
  "$schema": "https://json.schemastore.org/sarif-2.1.0-rtm.5.json",
  "runs": [
    {
      "tool": {
        "driver": {
          "informationUri": "https://deepsource.io",
          "name": "deepsource",
          "rules": [
            {
              "id": "RVV-B0013",
              "name": "Unused method receiver detected",
              "shortDescription": {
                "text": ""
              },
              "helpUri": "",
              "properties": {
                "category": "",
                "recommended": ""
              }
            }
          ]
        }
      },
      "results": [
        {
          "ruleId": "RVV-B0013",
          "ruleIndex": 0,
          "kind": "fail",
          "level": "error",
          "message": {
            "text": "Unused method receiver detected"
          },
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "deepsource/transformers/queries/get_transformers.go"
                },
                "region": {
                  "startLine": 34,
                  "endLine": 34
                }
              }
            }
          ]
        },
        {
          "ruleId": "RVV-B0013",
          "ruleIndex": 1,
          "kind": "fail",
          "level": "error",
          "message": {
            "text": "Unused method receiver detected"
          },
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "deepsource/transformers/queries/get_transformers.go"
                },
                "region": {
                  "startLine": 44,
                  "endLine": 44
                }
              }
            }
          ]
        }
      ]
    }
  ]
}
