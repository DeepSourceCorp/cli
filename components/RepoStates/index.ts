import BaseState from './BaseState.vue'
import RepoEmpty from './RepoEmpty.vue'
import RepoError from './RepoError.vue'
import RepoInactive from './RepoInactive.vue'
import RepoTimeout from './RepoTimeout.vue'
import RepoWaiting from './RepoWaiting.vue'

export { BaseState, RepoEmpty, RepoError, RepoInactive, RepoTimeout, RepoWaiting }
