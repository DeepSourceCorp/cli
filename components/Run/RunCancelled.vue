<template>
  <base-state title="Analysis cancelled">
    <template slot="hero">
      <empty-state-picture
        :webp-image-path="require('~/assets/images/ui-states/repo/run-cancelled.webp')"
        :png-image-path="require('~/assets/images/ui-states/repo/run-cancelled.png')"
        width="w-26"
        class="mx-auto mb-4"
        alt-text="Analysis cancelled"
      />
    </template>
    <span class="text-vanilla-400"
      >This analysis run did not complete and was cancelled due to an error.</span
    >
  </base-state>
</template>
<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator'
import { BaseState } from '@/components/RepoStates'

/**
 * ? Cancelled empty state for Analyzer runs.
 */
@Component({ components: { BaseState } })
export default class RunCancelled extends Vue {}
</script>
