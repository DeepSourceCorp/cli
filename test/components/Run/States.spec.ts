import '@testing-library/jest-dom'
import { render } from '@testing-library/vue'
import { createLocalVue, mount } from '@vue/test-utils'
import vuex from 'vuex'

import { BaseState, RepoEmpty, RepoError, RepoTimeout, RepoWaiting } from '~/components/RepoStates'
import RunFailed from '~/components/Run/RunFailed.vue'
import RunLoading from '~/components/Run/RunLoading.vue'
import RunNuked from '~/components/Run/RunNuked.vue'
import RunPass from '~/components/Run/RunPass.vue'
import RunTimeout from '~/components/Run/RunTimeout.vue'
import RunWaiting from '~/components/Run/RunWaiting.vue'
import RunCancelled from '~/components/Run/RunCancelled.vue'

import { storeModulesGenerator } from '~/test/mocks'

const stubs = {
  ZButton: true,
  ZIcon: true,
  EmptyState: true,
  EmptyStatePicture: true
}

const props = {
  id: 'UmVwb3NpdG9yeTp6dmp2eXo=',
  name: 'asgard',
  defaultBranchName: 'master'
}

jest.mock('~/assets/images/ui-states/repo/error.svg', () => 'error.svg')
jest.mock('~/assets/images/ui-states/repo/empty.svg', () => 'empty.svg')
jest.mock('~/assets/images/ui-states/repo/timeout.svg', () => 'timeout.svg')

describe('[[ Run Status Components ]]', () => {
  test('RunCancelled', () => {
    const { html } = render(RunCancelled, { stubs })
    expect(html()).toMatchSnapshot('RunCancelled')
  })
  test('RunTimeout', () => {
    const { html } = render(RunTimeout, { stubs })
    expect(html()).toMatchSnapshot('RunTimeout')
  })
  test('RunLoading', () => {
    const { html } = render(RunLoading, { stubs })
    expect(html()).toMatchSnapshot('RunLoading')
  })
  test('RunNuked', () => {
    const { html } = render(RunNuked, { stubs })
    expect(html()).toMatchSnapshot('RunNuked')
  })
  test('RunPass', () => {
    const { html } = render(RunPass, { stubs })
    expect(html()).toMatchSnapshot('RunPass')
  })
  test('RunWaiting', () => {
    const { html } = render(RunWaiting, { stubs })
    expect(html()).toMatchSnapshot('RunWaiting')
  })
  test('RunFailed', () => {
    const { html } = render(RunFailed, { stubs })
    expect(html()).toMatchSnapshot('RunFailed')
  })
})

describe('[[ Repo Status Components ]]', () => {
  const renderOptions = {
    components: {
      BaseState
    },
    props,
    stubs
  }

  test('RepoTimeout', () => {
    const mocks = {
      $toast: {
        success: jest.fn()
      }
    }
    const localVue = createLocalVue()
    localVue.use(vuex)

    const div = document.createElement('div')
    document.body.appendChild(div)

    const wrapper = mount(RepoTimeout, {
      stubs,
      propsData: props,
      localVue,
      attachTo: div,
      mocks,
      store: new vuex.Store({
        modules: storeModulesGenerator({
          'repository/detail': {
            namespaced: true,
            actions: {
              toggleRepoActivation: jest.fn()
            }
          }
        })
      })
    })

    expect(wrapper.html()).toMatchSnapshot('RepoTimeout')
  })

  test('RepoEmpty', () => {
    const { html } = render(RepoEmpty, renderOptions)
    expect(html()).toMatchSnapshot('RepoEmpty')
  })

  test('RepoError', () => {
    const { html } = render(RepoError, renderOptions)
    expect(html()).toMatchSnapshot('RepoError')
  })

  test('RepoWaiting', () => {
    const { html } = render(RepoWaiting, renderOptions)
    expect(html()).toMatchSnapshot('RepoWaiting')
  })
})
