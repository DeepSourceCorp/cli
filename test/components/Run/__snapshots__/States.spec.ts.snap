// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[[ Repo Status Components ]] RepoEmpty: RepoEmpty 1`] = `
"<section class=\\"grid border border-dashed rounded-md border-slate-400 place-content-center min-h-92 max-h-screen p-12\\" id=\\"UmVwb3NpdG9yeTp6dmp2eXo=\\" name=\\"asgard\\" defaultbranchname=\\"master\\"><img src=\\"timeout.svg\\" alt=\\"Repo Empty\\" class=\\"mx-auto mb-4\\">
  <h1 class=\\"text-base font-semibold text-center text-vanilla-100 mt-6\\">
    No code found in the repository
  </h1>
  <div class=\\"max-w-xl mt-2 text-center text-vanilla-400 text-sm\\">
    This repository seems to be empty. Please add code and check back later.
  </div>
</section>"
`;

exports[`[[ Repo Status Components ]] RepoError: RepoError 1`] = `
"<section class=\\"grid border border-dashed rounded-md border-slate-400 place-content-center min-h-92 max-h-screen p-12\\" id=\\"UmVwb3NpdG9yeTp6dmp2eXo=\\" name=\\"asgard\\" defaultbranchname=\\"master\\"><img src=\\"timeout.svg\\" alt=\\"Repo Error\\" class=\\"mx-auto mb-4\\">
  <h1 class=\\"text-base font-semibold text-center text-vanilla-100 mt-6\\">
    Analysis has been halted
  </h1>
  <div class=\\"max-w-xl mt-2 text-center text-vanilla-400 text-sm\\">
    <p>
      We have been unable to analyze the code due to issues with analysis configuration. These
      issues must be resolved before we can continue.
    </p>
  </div>
</section>"
`;

exports[`[[ Repo Status Components ]] RepoTimeout: RepoTimeout 1`] = `
"<emptystate-stub use-v2=\\"\\" show-border=\\"\\" webp-image-path=\\"timeout.svg\\" png-image-path=\\"timeout.svg\\" title=\\"Analysis timed out\\" subtitle=\\"We had to cancel this analysis as it took too long to finish. Try making another commit on this branch to retry.\\" width=\\"w-26\\" content-width=\\"max-w-sm\\" defaultbranchname=\\"master\\">
  <p class=\\"max-w-xl\\">
    We were not able to finish the analysis in time. This can happen if there are too many files
    to analyze. Please give it another shot.
  </p><button href=\\"#\\" type=\\"button\\" class=\\"inline-flex items-center font-medium transition-colors duration-300 ease-in-out rounded-sm z-btn focus:outline-none whitespace-nowrap mt-4 text-ink-400 justify-center p-0 h-8 px-4 py-1 text-xs space-x-1 leading-loose bg-juniper hover:bg-juniper-600\\" data-testid=\\"activate-analysis\\">
    <zicon-stub icon=\\"repeat\\" size=\\"small\\" position=\\"\\" color=\\"ink-400\\" class=\\"mr-1.5\\"></zicon-stub>
    Re-run analysis
  </button>
</emptystate-stub>"
`;

exports[`[[ Repo Status Components ]] RepoWaiting: RepoWaiting 1`] = `
"<section class=\\"grid border border-dashed rounded-md border-slate-400 place-content-center min-h-92 max-h-screen p-12\\" id=\\"UmVwb3NpdG9yeTp6dmp2eXo=\\" name=\\"asgard\\" defaultbranchname=\\"master\\"><video poster=\\"timeout.svg\\" autoplay=\\"autoplay\\" loop=\\"loop\\" disablepictureinpicture=\\"\\" muted=\\"muted\\" class=\\"mx-auto w-26\\">
    <source src=\\"timeout.svg\\" type=\\"video/webm\\"></video>
  <h1 class=\\"text-base font-semibold text-center text-vanilla-100 mt-6\\">
    Waiting for results
  </h1>
  <div class=\\"max-w-xl mt-2 text-center text-vanilla-400 text-sm\\">
    <p class=\\"max-w-lg\\">
      This is the first analysis for this repository. We're analyzing all the files, so it can take
      a while.
    </p>
  </div>
</section>"
`;

exports[`[[ Run Status Components ]] RunCancelled: RunCancelled 1`] = `
"<section class=\\"grid border border-dashed rounded-md border-slate-400 place-content-center min-h-92 max-h-screen p-12\\">
  <emptystatepicture-stub webp-image-path=\\"timeout.svg\\" png-image-path=\\"timeout.svg\\" width=\\"w-26\\" alt-text=\\"Analysis cancelled\\" class=\\"mx-auto mb-4\\"></emptystatepicture-stub>
  <h1 class=\\"text-base font-semibold text-center text-vanilla-100 mt-6\\">
    Analysis cancelled
  </h1>
  <div class=\\"max-w-xl mt-2 text-center text-vanilla-400 text-sm\\"> <span class=\\"text-vanilla-400\\">This analysis run did not complete and was cancelled due to an error.</span></div>
</section>"
`;

exports[`[[ Run Status Components ]] RunFailed: RunFailed 1`] = `
"<section class=\\"grid border border-dashed rounded-md border-slate-400 place-content-center min-h-92 max-h-screen p-12\\">
  <emptystatepicture-stub webp-image-path=\\"timeout.svg\\" png-image-path=\\"timeout.svg\\" width=\\"w-26\\" alt-text=\\"Run failed\\" class=\\"mx-auto mb-4\\"></emptystatepicture-stub>
  <h1 class=\\"text-base font-semibold text-center text-vanilla-100 mt-6\\">
    Run failed because of a failing metric
  </h1>
  <div class=\\"max-w-xl mt-2 text-center text-vanilla-400 text-sm\\"></div>
</section>"
`;

exports[`[[ Run Status Components ]] RunLoading: RunLoading 1`] = `
"<section class=\\"grid border border-dashed rounded-md border-slate-400 place-content-center min-h-92 max-h-screen p-12\\"><video poster=\\"timeout.svg\\" autoplay=\\"autoplay\\" loop=\\"loop\\" disablepictureinpicture=\\"\\" muted=\\"muted\\" class=\\"mx-auto w-26\\">
    <source src=\\"timeout.svg\\" type=\\"video/webm\\"></video>
  <h1 class=\\"text-base font-semibold text-center text-vanilla-100 mt-6\\">
    Waiting for results
  </h1>
  <div class=\\"max-w-xl mt-2 text-center text-vanilla-400 text-sm\\">
    <p class=\\"max-w-md\\">
      We're currently analyzing the branch, it usually takes a few minutes to fetch the results.
    </p>
  </div>
</section>"
`;

exports[`[[ Run Status Components ]] RunNuked: RunNuked 1`] = `"<emptystate-stub webp-image-path=\\"timeout.svg\\" png-image-path=\\"timeout.svg\\" title=\\"Artifacts not received in time\\" subtitle=\\"We had to cancel this analysis as we didn't receive additional artifacts or reports required by the Analyzer in time. Try making another commit on this branch to retry.\\"></emptystate-stub>"`;

exports[`[[ Run Status Components ]] RunPass: RunPass 1`] = `"<emptystate-stub webp-image-path=\\"timeout.svg\\" png-image-path=\\"timeout.svg\\" title=\\"No issues found in this check\\" subtitle=\\"Give yourself a cookie.\\"></emptystate-stub>"`;

exports[`[[ Run Status Components ]] RunTimeout: RunTimeout 1`] = `"<emptystate-stub webp-image-path=\\"timeout.svg\\" png-image-path=\\"timeout.svg\\" title=\\"Analysis timed out\\" subtitle=\\"We had to cancel this analysis as it took too long to finish. Try making another commit on this branch to retry.\\"></emptystate-stub>"`;

exports[`[[ Run Status Components ]] RunWaiting: RunWaiting 1`] = `
"<section class=\\"grid border border-dashed rounded-md border-slate-400 place-content-center min-h-92 max-h-screen p-12\\"><video poster=\\"timeout.svg\\" autoplay=\\"autoplay\\" loop=\\"loop\\" disablepictureinpicture=\\"\\" muted=\\"muted\\" class=\\"mx-auto w-26\\">
    <source src=\\"timeout.svg\\" type=\\"video/webm\\"></video>
  <h1 class=\\"text-base font-semibold text-center text-vanilla-100 mt-6\\">
    Waiting to receive artifacts
  </h1>
  <div class=\\"max-w-xl mt-2 text-center text-vanilla-400 text-sm\\">
    <p>
      The analysis will begin as soon as we receive additional artifacts or reports required by the
      Analyzer.
    </p>
  </div>
</section>"
`;
